frontend/learning/components/ItemDisplay/WordDisplay/StrokeGif.js

```diff
@@ -0,0 +1,127 @@
+
+import HanziWriter from 'hanzi-writer';
+import React, {useEffect, useState} from 'react';
+import styled from 'styled-components';
+import '@learning.styles/ItemDisplay.css';
+import PropTypes from 'prop-types';
+
+const WordContainer = styled.div`
+    font-size: 3.75 em;
+    font-weight: 200;
+    text-align: center;
+    display: flex;
+    flex-direction: row;
+`;
+
+/**
+ * @param {String} props.item
+ * @param {Object} props
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export default function StrokeGif(props) {
+
+    /* Split input into array of characters. */
+    const items = props.item.split('');
+
+    /* Create target ID for each of the characters. */
+    const itemsTargetIDs = items.map((value, index) => `writer-target-${index}`);
+
+    // an array of 'writers'
+    let writers = null; 
+
+    const [gifState1, setGifState1] = useState(0);
+    const [gifState2, setGifState2] = useState(0);
+    /* 1 - Minor Problem: How to create an array of states?
+    const [state, setState] = itemsTargetIDs.map((value, index) => {
+        [state[index], setState[index]] = useState(0);
+    }); 
+    */
```

Comment:
reference CND review questions for detail, be careful that you can only modify states with setState, so things like modifying an array element takes a bit of extra work. a cheat sheet: https://dev.to/andyrewlee/cheat-sheet-for-updating-objects-and-arrays-in-react-state-48np

Reply:
https://github.com/solved-chinese/app/blob/development/frontend/learning/components/ReviewQuestion/ClickAndDrop.js